src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -418,13 +428,21 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
                 $markingStoreDefinition = new Reference($workflow['marking_store']['service']);
             }
 
-            $workflowDefinition = new DefinitionDecorator('workflow.abstract');
+            $definitionDefinition->addTag('workflow.definition', array(
+                'name' => $name,
+                'type' => $type,
+                'marking_store' => isset($workflow['marking_store']['type']) ? $workflow['marking_store']['type'] : null,
+            ));
+            $definitionDefinition->setPublic(false);
```

Comment:
I've been playing with this implementation a bit. I think I would like to have the definition to be public. Otherwise I can't dump my graph not validate it again (for whatever reason)